OrbitGl/TimeGraph.cpp

```diff
@@ -726,41 +720,34 @@ void TimeGraph::SortTracks() {
   }
 }
 
-//----------------------------------------------------------------------------
-void TimeGraph::OnLeft() {
+const TextBox* TimeGraph::FindPrevious() {
   TextBox* selection = Capture::GSelectedTextBox;
+  const TextBox* previous = nullptr;
   if (selection) {
     const Timer& timer = selection->GetTimer();
-    const TextBox* left = nullptr;
     if (timer.m_Type == Timer::GPU_ACTIVITY) {
-      left = GetOrCreateGpuTrack(GetGpuTimelineHash(timer))->GetLeft(selection);
+      previous =
+          GetOrCreateGpuTrack(GetGpuTimelineHash(timer))->GetLeft(selection);
     } else {
-      left = GetOrCreateThreadTrack(timer.m_TID)->GetLeft(selection);
-    }
-    if (left) {
-      SelectLeft(left);
+      previous = GetOrCreateThreadTrack(timer.m_TID)->GetLeft(selection);
     }
   }
-  NeedsUpdate();
+  return previous;
 }
 
-//----------------------------------------------------------------------------
-void TimeGraph::OnRight() {
+const TextBox* TimeGraph::FindNext() {
```

Comment:
Maybe FindNextOfSelected/FindPreviousOfSelected? I'm not sure about that, so feel free to ignore it, if you don't like it.

Reply:
+1

Or pass in the textbox as starting point.